src/graph/graph.rs

```diff
@@ -56,20 +56,18 @@ impl Graph {
         self.edges
             .iter()
             .fold(HashMap::new(), |mut rules, (from_vertex_index, edges)| {
```

Comment:
`edges` is actually quite a confusing name in the context of this, had to go back to the Edges structure to understand what was going on. Maybe we can rename the structure we're iterating over to `connections`

Reply:
Not in this PR but will change.